src/ArmTemplates/Extractor/EntityExtractors/ParametersExtractor.cs

```diff
@@ -119,6 +120,39 @@ async Task AddServiceUrlParameterAsync()
                 parameters.Add(ParameterNames.ServiceUrl, new TemplateObjectParameterProperties() { Value = serviceUrls });
             }
 
+            async Task AddApiOauth2ScopeParameterAsync()
+            {
+                if (!extractorParameters.ParametrizeApiOauth2Scope)
+                {
+                    return;
+                }
+
+                var apiOauth2Scopes = new Dictionary<string, string>();
+                foreach (var apiName in apisToExtract)
+                {
+                    var apiDetails = await this.apisClient.GetSingleAsync(apiName, extractorParameters);
+
+                    if (apiDetails.Properties.AuthenticationSettings?.OAuth2 is not null)
+                    {
+                        string apiOAuthScope;
+                        var validApiName = ParameterNamingHelper.GenerateValidParameterName(apiName, ParameterPrefix.Api);
+
+                        if (extractorParameters.ApiOauth2ScopeParameters is null)
+                        {
+                            apiOAuthScope = apiDetails.Properties.AuthenticationSettings.OAuth2?.Scope;
+                        }
+                        else
+                        {
+                            apiOAuthScope = extractorParameters.ApiOauth2ScopeParameters.FirstOrDefault(x => x.ApiName.Equals(apiName))?.Scope;
```

Comment:
If you are trying to find appropriate api based on `ApiName` in ApiOAuth2ScopeParameters collection, maybe we can make it of dictionary type, so that search goes faster here? (probably `Dictionary<string (apiName), ApiOauth2ScopeProperty (scope definition)>`)

I know we are not pursuing speed and performance in this application, but in case smb has 1000 apis, it will take a lot of time to iterate through array multiple times and find the apiName

Reply:
As per discussion: New apiParameter configuration introduced for extractor which allows configure service url and oauthscope 2 of apis in one Dictionary. For example please see the Updated documentation.